### PR TITLE
Added check that scan goes from -x to x

### PR DESCRIPTION
**Pull request by windelbouwman**

### CODE REVIEW

**Comment by windelbouwman** [comment]
See this issue:

https://github.com/ros-perception/slam_gmapping/issues/25

File: gmapping/src/slam_gmapping.cpp [hunk 1]
```diff
@@ -309,6 +309,13 @@ SlamGMapping::initMapper(const sensor_msgs::LaserScan& scan)
     return false;
   }
 
+  // Check that laserscan is from -x to x in angles:
+  if (fabs(fabs(scan.angle_min) - fabs(scan.angle_max)) > 0.001)
```
Comment (wjwwood) [inline gmapping/src/slam_gmapping.cpp]: Can you use `FLT_EPSILON` from `<cfloat>` here? See:

http://www.cplusplus.com/reference/cfloat/

**Comment by wjwwood** [comment]
Thanks for the pull request.

So before, what was the failure state? It worked but just gave the wrong result or did it crash or something later on?

**Comment by windelbouwman** [comment]
There was no crash, just plain wrong results. I think the gmapping library assumes scans to be centered around 0 angle.

File: gmapping/src/slam_gmapping.cpp (hunk 1 above)
Comment (windelbouwman) [inline gmapping/src/slam_gmapping.cpp]: Check.

**Comment by wjwwood** [comment]
Ok, well I'll merge this, but you should `#include <cfloat>` at the top of the file to make sure the `FLT_EPSILON` is defined.

**Comment by windelbouwman** [comment]
Good point! I added the include.

**Pull request merged**